flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/preprocessing/MinMaxScaler.scala

```diff
@@ -0,0 +1,254 @@
+import org.apache.flink.ml.pipeline.{FitOperation, TransformOperation, Transformer}
+import org.apache.flink.ml.preprocessing.MinMaxScaler.{Max, Min}
+
+import scala.reflect.ClassTag
+
+/** Scales observations, so that all features are in a user-specified range.
+  * By default for [[MinMaxScaler]] transformer range = [0,1].
+  *
+  * This transformer takes a subtype of  [[Vector]] of values and maps it to a
+  * scaled subtype of [[Vector]] such that each feature lies between a user-specified range.
+  *
+  * This transformer can be prepended to all [[Transformer]] and
+  * [[org.apache.flink.ml.pipeline.Predictor]] implementations which expect as input a subtype
+  * of [[Vector]].
+  *
+  * @example
+  * {{{
+  *               val trainingDS: DataSet[Vector] = env.fromCollection(data)
+  *               val transformer = MinMaxScaler().setMin(-1.0)
+  *
+  *               transformer.fit(trainingDS)
+  *               val transformedDS = transformer.transform(trainingDS)
+  * }}}
+  *
+  * =Parameters=
+  *
+  * - [[Min]]: The minimum value of the range of the transformed data set; by default equal to 0
+  * - [[Max]]: The maximum value of the range of the transformed data set; by default
+  * equal to 1
+  */
+class MinMaxScaler extends Transformer[MinMaxScaler] {
+
+  var metricsOption: Option[DataSet[(linalg.Vector[Double], linalg.Vector[Double])]] = None
+
+  /** Sets the minimum for the range of the transformed data
+    *
+    * @param min the user-specified minimum value.
+    * @return the MinMaxScaler instance with its minimum value set to the user-specified value.
+    */
+  def setMin(min: Double): MinMaxScaler = {
+    parameters.add(Min, min)
+    this
+  }
+
+  /** Sets the maximum for the range of the transformed data
+    *
+    * @param max the user-specified maximum value.
+    * @return the MinMaxScaler instance with its minimum value set to the user-specified value.
+    */
+  def setMax(max: Double): MinMaxScaler = {
+    parameters.add(Max, max)
+    this
+  }
+}
+
+object MinMaxScaler {
+
+  // ====================================== Parameters =============================================
+
+  case object Min extends Parameter[Double] {
+    override val defaultValue: Option[Double] = Some(0.0)
+  }
+
+  case object Max extends Parameter[Double] {
+    override val defaultValue: Option[Double] = Some(1.0)
+  }
+
+  // ==================================== Factory methods ==========================================
+
+  def apply(): MinMaxScaler = {
+    new MinMaxScaler()
+  }
+
+  // ====================================== Operations =============================================
+
+  /** Trains the [[org.apache.flink.ml.preprocessing.MinMaxScaler]] by learning the minimum and
+    * maximum of each feature of the training data. These values are used in the transform step
+    * to transform the given input data.
+    *
+    * @tparam T Input data type which is a subtype of [[Vector]]
+    * @return
```

Comment:
Well the function's return type is a `FitOperation`. Thus, we should add a description for the return statement. Will do it when merging.